src/vivarium/framework/randomness/index_map.py

```diff
@@ -134,34 +133,36 @@ def convert_to_ten_digit_int(self, column: pd.Series) -> pd.Series:
         return column
 
     @staticmethod
-    def digit(m: Union[int, pd.Series], n: int) -> Union[int, pd.Series]:
+    def _digit(m: Union[int, pd.Series], n: int) -> Union[int, pd.Series]:
         """Returns the nth digit of each number in m."""
         return (m // (10**n)) % 10
 
     @staticmethod
-    def clip_to_seconds(m: Union[int, pd.Series]) -> Union[int, pd.Series]:
+    def _clip_to_seconds(m: Union[int, pd.Series]) -> Union[int, pd.Series]:
         """Clips UTC datetime in nanoseconds to seconds."""
         return m // pd.Timedelta(1, unit="s").value
 
-    def spread(self, m: Union[int, pd.Series]) -> Union[int, pd.Series]:
+    def _spread(self, m: Union[int, pd.Series]) -> Union[int, pd.Series]:
         """Spreads out integer values to give smaller values more weight."""
         return (m * 111_111) % self.TEN_DIGIT_MODULUS
 
-    def shift(self, m: Union[float, pd.Series]) -> Union[int, pd.Series]:
+    def _shift(self, m: Union[float, pd.Series]) -> Union[int, pd.Series]:
         """Shifts floats so that the first 10 decimal digits are significant."""
         out = m % 1 * self.TEN_DIGIT_MODULUS // 1
         if isinstance(out, pd.Series):
             return out.astype("int64")
         return int(out)
 
     def __getitem__(self, index: pd.Index) -> pd.Series:
+        if not self._use_crn:
+            return pd.Series(index, index=index)
         if isinstance(index, (pd.Index, pd.MultiIndex)):
             return self._map[index]
         else:
             raise IndexError(index)
 
     def __len__(self) -> int:
-        return len(self._map)
+        return self._size
```

Comment:
Where is `len` called on an `IndexMap` besides `randomness_stream.random()`?

Reply:
It's not.